lib/v2/gov/cac/index.js

```diff
@@ -0,0 +1,55 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+const timezone = require('@/utils/timezone');
+
+module.exports = async (ctx) => {
+    const path = ctx.params[0];
+    const host = 'http://www.cac.gov.cn';
+    const homepage = `${host}/index.htm`;
+    const { data: homepageResponse } = await got(homepage);
+    const $ = cheerio.load(homepageResponse);
+    // 在首页查找出所有的目录完整路径，比如http://www.cac.gov.cn/xxh/A0906index_1.htm
+    // xxh --> {"path": "/xxh", "pathWithHtmlSubfix": "/xxh/A0906index_1.htm", "completeUrl": "http://www.cac.gov.cn/xxh/A0906index_1.htm"}
+    const pathList = $('a')
+        .toArray()
+        .map((item) => {
+            const href = $(item).attr('href');
+            if (href && href.startsWith('http://www.cac.gov.cn/')) {
+                const matchArray = href.match(/http:\/\/www\.cac\.gov\.cn(.*?)\/(A.*?\.htm)/);
+                if (matchArray && matchArray.length > 2) {
+                    const path = matchArray[1];
+                    const htmlName = matchArray[2];
+                    return {
+                        path,
+                        completeUrl: `${host}${path}/${htmlName}`,
+                    };
+                }
+            }
+            return undefined;
+        });
```

Comment:
The `pathList` here will mostly be the same, so it's better to cache it instead of regenerating it again.
```js
const config = require('@/config').value;
```
```suggestion
    // 在首页查找出所有的目录完整路径，比如http://www.cac.gov.cn/xxh/A0906index_1.htm
    // xxh --> {"path": "/xxh", "pathWithHtmlSubfix": "/xxh/A0906index_1.htm", "completeUrl": "http://www.cac.gov.cn/xxh/A0906index_1.htm"}
    const pathList = await ctx.cache.tryGet(
        'gov:cac:pathList',
        async () => {
            const { data: homepageResponse } = await got(homepage);
            const $ = cheerio.load(homepageResponse);
            return $('a')
                .toArray()
                .map((item) => {
                    const href = $(item).attr('href');
                    if (href && href.startsWith('http://www.cac.gov.cn/')) {
                        const matchArray = href.match(/http:\/\/www\.cac\.gov\.cn(.*?)\/(A.*?\.htm)/);
                        if (matchArray && matchArray.length > 2) {
                            const path = matchArray[1];
                            const htmlName = matchArray[2];
                            return {
                                path,
                                completeUrl: `${host}${path}/${htmlName}`,
                            };
                        }
                    }
                    return null;
                })
                .filter((item) => item);
        },
        config.cache.routeExpire,
        false
    );
```